pkg/client/cache/listers.go

```diff
@@ -135,7 +135,9 @@ func (s storeToNodeConditionLister) List() (nodes api.NodeList, err error) {
 		// Get the last condition of the required type
 		for _, cond := range node.Status.Conditions {
 			if cond.Type == s.conditionType {
-				nodeCondition = &cond
+				condCopy := cond
+				nodeCondition = &condCopy
+				break
```

Comment:
I wasn't sure about the correct behavior either, so I went with the same change that made it into upstream here: https://github.com/kubernetes/kubernetes/commit/a2570926da1b87856351c783c908c255042a9193#diff-53f28720440ab4b589ef2273346cecfcR139

But per the code comment above this block, I would expect to drop this break and have it use the last matching condition.